skeleton.py

```diff
@@ -564,124 +564,155 @@ def toDB(self, clearUpdateTag=False):
 		def txnUpdate(key, mergeFrom, clearUpdateTag):
 			blobList = set()
 			skel = type(mergeFrom)()
+
 			# Load the current values from Datastore or create a new, empty db.Entity
 			if not key:
 				dbObj = db.Entity(skel.kindName)
 				oldBlobLockObj = None
+
 			else:
 				k = db.Key(key)
 				assert k.kind() == skel.kindName, "Cannot write to invalid kind!"
+
 				try:
 					dbObj = db.Get(k)
+
 				except db.EntityNotFoundError:
 					dbObj = db.Entity(k.kind(), id=k.id(), name=k.name(), parent=k.parent())
+
 				else:
 					skel.setValues(dbObj)
+
 				try:
 					oldBlobLockObj = db.Get(db.Key.from_path("viur-blob-locks", str(k)))
+
 				except:
 					oldBlobLockObj = None
 
-			# Remember old hashes for bones that must have an unique value
-			oldUniqeValues = {}
-			for boneName, boneInstance in skel.items():
-				if boneInstance.unique:
-					if "%s.uniqueIndexValue" % boneName in dbObj:
-						oldUniqeValues[boneName] = dbObj["%s.uniqueIndexValue" % boneName]
-
-			## Merge the values from mergeFrom in
+			# Merge values and assemble unique properties
+			oldUniqueValues = {}
 			for key, bone in skel.items():
+				# Remember old hashes for bones that must have an unique value
+				if bone.unique:
+					if "%s.uniqueIndexValue" % key in dbObj:
+						oldUniqueValues[key] = dbObj["%s.uniqueIndexValue" % key]
+
+				# Merge the values from mergeFrom in
 				if key in mergeFrom:
 					bone.mergeFrom(skel.valuesCache, key, mergeFrom)
-			for key, _bone in skel.items():
-				dbObj = _bone.serialize(skel.valuesCache, key, dbObj)
-				blobList.update(_bone.getReferencedBlobs(self.valuesCache, key))
+
+				# Serialize bone into entity
+				dbObj = bone.serialize(skel.valuesCache, key, dbObj)
+
+				# Obtain referenced blobs
+				if key in self.valuesCache:
```

Comment:
This is not identical to the old logic. Why should we make assumptions about the layout inside the entity here? We simple can call getReferencedBlobs and let the bone sort things out

Reply:
Do you mean that `bone.getReferencedBlobs()` should check for its own bone existence?
This is OK for me, but I know that this was the causing for several issues in the past, when a bone does not check for its keys existence within the current cache.

Reply:
I've fixed it now.